drivers/sensors/lepton.c

```diff
@@ -397,66 +397,75 @@ static int config(omv_csi_t *csi, omv_csi_config_t config) {
 static int snapshot(omv_csi_t *csi, image_t *image, uint32_t flags) {
     framebuffer_t *fb = csi->fb;
 
-    if (flags & OMV_CSI_CAPTURE_FLAGS_NBLOCK) {
-        if (!vospi_active()) {
-            // Start the capture without blocking
-            vospi_snapshot(0);
-            return OMV_CSI_ERROR_WOULD_BLOCK;
-        }
-
-        // If capture is running and no frames return.
-        if (!framebuffer_get_head(fb, FB_PEEK)) {
-            return OMV_CSI_ERROR_WOULD_BLOCK;
-        }
-    }
-
-    if (flags & OMV_CSI_CAPTURE_FLAGS_UPDATE) {
-        image_t tmp;
-        framebuffer_init_image(fb, &tmp);
-        framebuffer_update_jpeg_buffer(&tmp);
-    }
-
     if (csi->pixformat == PIXFORMAT_INVALID) {
         return OMV_CSI_ERROR_INVALID_PIXFORMAT;
     }
 
-    if (csi->framesize == OMV_CSI_FRAMESIZE_INVALID) {
+    if (!lepton.h_res || !lepton.v_res ||
+        csi->framesize == OMV_CSI_FRAMESIZE_INVALID) {
         return OMV_CSI_ERROR_INVALID_FRAMESIZE;
     }
 
-    if (!lepton.h_res || !lepton.v_res) {
-        return OMV_CSI_ERROR_INVALID_FRAMESIZE;
-    }
-
-    if (resolution[csi->framesize][0] < lepton.h_res || resolution[csi->framesize][1] < lepton.v_res) {
+    if (resolution[csi->framesize][0] < lepton.h_res ||
+        resolution[csi->framesize][1] < lepton.v_res) {
         return OMV_CSI_ERROR_INVALID_FRAMESIZE;
     }
 
     if (omv_csi_check_framebuffer_size(csi) == -1) {
         return OMV_CSI_ERROR_FRAMEBUFFER_OVERFLOW;
     }
 
-    for (int i = 0; i < LEPTON_SNAPSHOT_RETRY; i++) {
-        if (vospi_snapshot(LEPTON_SNAPSHOT_TIMEOUT) == 0) {
+    if (flags & OMV_CSI_CAPTURE_FLAGS_UPDATE) {
```

Comment:
Some of the if statements and the update flag here could be moved to common to become standard for all drivers.